### PR TITLE
Add support for custom managed secret types

### PR DESCRIPTION
**Pull request by nmanoogian**
This PR adds custom Kubernetes secret types to the operator. The README and docs changes describe the use cases and behavior changes.

Closes ENG-6130

### CODE REVIEW

**Comment by bjd2385** [comment]
Wow at first glance over the changes and doc, this is an amazing change/update. May do exactly what I needed

**Review by Piccirello — APPROVED** [review]
Nice docs!

File: controllers/dopplersecret_controller_secrets.go [hunk 1]
```diff
@@ -83,7 +79,10 @@ func (r *DopplerSecretReconciler) GetReferencedSecret(ctx context.Context, ref s
 
 // GetDopplerToken gets the Doppler Service Token referenced by the DopplerSecret
 func (r *DopplerSecretReconciler) GetDopplerToken(ctx context.Context, dopplerSecret secretsv1alpha1.DopplerSecret) (string, error) {
-	tokenSecret, err := r.GetReferencedSecret(ctx, dopplerSecret.Spec.TokenSecretRef)
+	tokenSecret, err := r.GetReferencedSecret(ctx, types.NamespacedName{
```
Comment (seslattery) [inline controllers/dopplersecret_controller_secrets.go:82]: nit:  it's a little clunky defining the struct in the function call.  I'd just pull it out as a variable on the line above and pass it in.

File: controllers/dopplersecret_controller_secrets.go [hunk 2]
```diff
@@ -214,10 +225,16 @@ func (r *DopplerSecretReconciler) UpdateSecret(ctx context.Context, dopplerSecre
 		return fmt.Errorf("Failed to load Doppler Token: %w", err)
 	}
 
-	existingKubeSecret, err := r.GetReferencedSecret(ctx, dopplerSecret.Spec.ManagedSecretRef)
+	existingKubeSecret, err := r.GetReferencedSecret(ctx, types.NamespacedName{
```
Comment (seslattery) [inline controllers/dopplersecret_controller_secrets.go:228]: nit: it's a little clunky defining the struct in the function call. I'd just pull it out as a variable on the line above and pass it in.

File: controllers/dopplersecret_controller_secrets.go [hunk 3]
```diff
@@ -181,7 +191,8 @@ func (r *DopplerSecretReconciler) CreateManagedSecret(ctx context.Context, doppl
 
 // UpdateManagedSecret updates a managed Kubernetes secret
 func (r *DopplerSecretReconciler) UpdateManagedSecret(ctx context.Context, secret corev1.Secret, dopplerSecret secretsv1alpha1.DopplerSecret, secretsResult models.SecretsResult) error {
-	secretData, dataErr := GetKubeSecretData(secretsResult, dopplerSecret.Spec.Processors)
+	includeSecretsByDefault := dopplerSecret.Spec.ManagedSecretRef.Type == string(corev1.SecretTypeOpaque)
```
Comment (seslattery) [inline controllers/dopplersecret_controller_secrets.go:194]: nit: This feels like it is not idiomatic. I think the more common paradigm is relying on a boolean's zero value, and then explicity setting the value if the condition passes. It is more verbose, but simpler to read.
```go
var includeSecretsByDefault bool
if  dopplerSecret.Spec.ManagedSecretRef.Type == string(corev1.SecretTypeOpaque) { 
  includeSecretsByDefault = true
}
```

**Review by seslattery — APPROVED** [review]
Looks good to me. I just have some minor nitpicks, but mergeable as is. I don't see any tests for this change, how do we test changes to the operator?

**Pull request merged**